cmd/pomerium/main.go

```diff
@@ -19,43 +20,65 @@ import (
 	"github.com/pomerium/pomerium/pkg/envoy/files"
 )
 
-var (
-	versionFlag = flag.Bool("version", false, "prints the version")
-	configFile  = flag.String("config", "", "Specify configuration file location")
-)
-
 func main() {
-	flag.Parse()
-	if *versionFlag {
-		fmt.Println("pomerium:", version.FullVersion())
-		fmt.Println("envoy:", files.FullVersion())
-		return
+	convertOldStyleFlags()
+	var configFile string
+	root := &cobra.Command{
+		Use:          "pomerium",
+		Version:      fmt.Sprintf("pomerium: %s\nenvoy: %s\n", version.FullVersion(), files.FullVersion()),
+		SilenceUsage: true,
 	}
+	root.AddCommand(zero_cmd.BuildRootCmd())
+	root.PersistentFlags().StringVar(&configFile, "config", "", "Specify configuration file location")
```

Comment:
Switching to cobra may be fine but I can't evaluate this for correctness. Pomerium has magical code when loading config.

Reply:
The only catch is that pflag is not backwards compatible with std flag, so "-config" is invalid, which is why there is code to handle this use case. From my testing this should be fully backwards compatible with any existing usages of the CLI (which are not many, there are only two possible flags and no positional args)

Reply:
It would get complex very quickly. There are also --token and --cluster-api-endpoint flags, and if we ever want to add more options or features to this (there was discussion about possibly adding zero client commands) it would become a burden to maintain logic to keep track of which flags can or can't be used with other flags, etc. Also maybe for consistency with pomerium-cli, and it does help with keeping the zero specific code logically separated.

If the concern is compatibility, I can add unit tests to validate, would that work?